### ISSUE TITLE
Key aliases missing after keycode-refactor, breaking user configs

### ISSUE THREAD

**Issue by Artefact2** [kmonad/kmonad#371]
As of 2356e27, kmonad will not start for me.

Here is the error:
~~~
Oct 05 16:33:36 Silmeria kmonad[1830]: kmonad: Mismatch between length of 'defsrc' and deflayer <default>
Oct 05 16:33:36 Silmeria kmonad[1830]: Source length: 2
Oct 05 16:33:36 Silmeria kmonad[1830]: Layer length: 1
~~~

Here is my config file:
~~~
(defcfg
        input (device-file "/dev/input/by-id/usb-TypeMatrix.com_USB_Keyboard-event-if01")
        output (uinput-sink "kmonad (typematrix 2030 106, media keys)")
        fallthrough true
        allow-cmd false)

(defsrc pp)
(deflayer default lmet)
~~~

**Comment by slotThe** [comment]
On Tue, Oct 05 2021 07:38, Romain D. wrote:
> (defsrc pp)

Is this a real key?  My guess is that it tries to parse this as two `p`s
because we have no key of that name anywhere that I can find

**Comment by Artefact2** [comment]
It used to be, see https://github.com/kmonad/kmonad/blob/7b5142b85660bec043f323ab32c4240a30332473/src/KMonad/Keyboard/Keycode.hs#L352

It seems a lot of the old abbreviations did not make it through in the new code.

**Comment by slotThe** [comment]
Ah I see.  Anything else in particular you are also missing?

**Comment by MuhammedZakir** [comment]
@slotThe is right.

I presume `pp` is referring to `play/pause` button? Currently, those aliases are not defined [1]. It was commented out in `keycode-refactor` branch, and @david-janssen probably forgot to re-add them.

[1] https://github.com/kmonad/kmonad/blob/2356e2734c029e9cf6af7781afce706fa00e8fa7/src/KMonad/App/Parser/Keycode.hs#L121

---

@david-janssen: Temporarily, we can at least add aliases for keycodes that are present in all 3 OSs. I did a quick-look, and the only problematic aliases are for `wakeup`, `launchpad` and `mission_control`.

**Comment by Artefact2** [comment]
Where can I find a list of all the "core names" ? I think a lot were left out, considering that the [typematrix template](https://github.com/kmonad/kmonad/blob/master/keymap/template/type_matrix_2030.kbd) badly fails too. (It has a few exotic keycodes like katakanahiragana, ro, etc.)

**Comment by MuhammedZakir** [comment]
> Where can I find a list of all the "core names" ? I think a lot were left out, considering that the [typematrix template](https://github.com/kmonad/kmonad/blob/master/keymap/template/type_matrix_2030.kbd) badly fails too. (It has a few exotic keycodes like katakanahiragana, ro, etc.)

Linux: https://github.com/kmonad/kmonad/blob/2356e2734c029e9cf6af7781afce706fa00e8fa7/src/KMonad/Util/Keyboard/Linux.hs#L119

**Comment by david-janssen** [comment]
D'oh!

I undid the merge, because I completely forgot about all the aliases. We'll tinker on the aliases in the keycode-refactor branch and write some tests so that all user configs parse without error, and then try again.

**Comment by slotThe** [comment]
I will close this because it's technically fixed :)  (The issue is mentioned in #372 so we definitely won't forget it)
